### PR TITLE
Add AltMode to Origin

### PR DESCRIPTION
**Pull request by JFranklin3000**
Closes [issue URL]

##### In this PR
- Created Option to add AltMode to Origin
- When and Origin with an AltMode is added to an actor canTransform is set to true
- deleting the Origin removes the altMode and if no remaining altModes sets canTransform to false
- fixes dataModel issue with Origin Skills and Essences

##### Testing
- Create Origin with and without altMode and drop on various actors

### CODE REVIEW

File: module/helpers/utils.mjs [hunk 1]
```diff
@@ -67,7 +67,6 @@ export function getShiftedSkill(skill, shift, actor) {
   let skillString = "";
   let currentShift = "";
   let newShift = "";
-
```
Comment (phildominguez) [inline module/helpers/utils.mjs:70]: Revert

File: module/sheet-handlers/background-handler.mjs [hunk 2]
```diff
@@ -289,11 +290,20 @@ export async function onOriginDelete(actor, origin) {
   const [newShift, skillString] = getShiftedSkill(selectedSkill, -1, actor);
   await deleteAttachmentsForItem(origin, actor);
 
+  let hasAltMode = false;
+  for (const [, item] of Object.entries(actor.items)) {
+    if (item.type == "altMode") {
+      hasAltMode = true;
+      break;
+    }
+  }
```
Comment (phildominguez) [inline module/sheet-handlers/background-handler.mjs:299]:
```suggestion
  const hasAltMode = !!getItemsOfType(actor.items).length;
```
I think this should work. Doing `!!` is an easy way of converting basic types into a bool.

File: module/helpers/utils.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/helpers/utils.mjs]: fixed

File: module/sheet-handlers/attachment-handler.mjs [hunk 3]
```diff
@@ -183,7 +188,9 @@ export async function setEntryAndAddItem(droppedItem, targetItem) {
 
     break;
   case "origin":
-    if (droppedItem.type == "perk") {
+    if (droppedItem.type =="altMode") {
+      return _addItemIfUnique(droppedItem, targetItem, entry);
+    } else if (droppedItem.type == "perk") {
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/attachment-handler.mjs]: changed

File: module/sheet-handlers/background-handler.mjs (hunk 2 above)
Comment (JFranklin3000) [reply module/sheet-handlers/background-handler.mjs]: changed and tested

**Review by phildominguez — APPROVED** [review]

**Pull request merged**